util/migration-rocksdb/src/lib.rs

```diff
@@ -239,7 +239,7 @@ impl Manager {
 			return Err(other_io_err("Migration impossible"));
 		};
 
-		let columns = migrations.get(0).and_then(|m| m.pre_columns());
+		let columns = migrations.get(0).map(|m| m.pre_columns()).unwrap_or(1);
```

Comment:
can you explain why we use 1 here? (could be map_or instead map().unwrap_or()`

However, because migrations have at least one element and `fn pre_columns() -> u32` this should never fail?

Reply:
My thinking was a migration of a database does not make much sense, so we know we have at least one column. Before this PR there could be one column and also the "default" (i.e. `None`).

Reply:
> My thinking was a migration of a zero-column database does not make much sense, so we know we have at least one column. Before this PR there could be one column and also the "default" (i.e. None).

Yeah, that makes sense.

My other comment was fuzzy sorry. I was referring to that  we check `if migrations.is_empty()` just before so we know that `migratations.get(0)` will always succeed and then after your changes `fn pre_colums` always return `u32` then I think we could use `expect` or just `migrations[0].pre_columns` because the `None` case never can happen?

Reply:
Got it, good catch.